migrator/migrations/m_180_to_m_181_group_role_permission_with_access_one/migration_impl.go

```diff
@@ -0,0 +1,83 @@
+package m180tom181
+
+import (
+	"context"
+
+	"github.com/stackrox/rox/generated/storage"
+	permissionsetstore "github.com/stackrox/rox/migrator/migrations/m_180_to_m_181_group_role_permission_with_access_one/permissionsetstore"
+	"github.com/stackrox/rox/migrator/types"
+	"github.com/stackrox/rox/pkg/sac"
+)
+
+const (
+	batchSize = 500
+)
+
+// Role: Replaced resource
+const (
+	Role = "Role"
+)
+
+// Access: Replacement resource
+const (
+	Access = "Access"
+)
+
+var (
+	replacements = map[string]string{
+		Role: Access,
+	}
+)
+
+func propagateAccessForPermission(permission string, accessLevel storage.Access, permissionSet map[string]storage.Access) storage.Access {
+	oldLevel, found := permissionSet[permission]
+	if !found {
+		return accessLevel
+	}
+	if accessLevel > oldLevel {
+		return oldLevel
+	}
+	return accessLevel
+}
+
+func migrate(database *types.Databases) error {
+	ctx := sac.WithAllAccess(context.Background())
+	store := permissionsetstore.New(database.PostgresDB)
+
+	migratedPermissionSets := make([]*storage.PermissionSet, 0, batchSize)
+	err := store.Walk(ctx, func(obj *storage.PermissionSet) error {
+		changed := false
```

Comment:
super-nit: moving the changed closer to where it's actually used instead of the top makes it a bit more clear reading through, was wondering initially where exactly this one will be used.